### PR TITLE
remove comments from config

### PR DESCRIPTION
**Pull request by charlierudolph**
@kevgo

What do you think of this? Strip yaml of comments and provide a link to the doc explaining them. The link will be based on the version when you generated the config.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+### allowUnused
+array of strings or regular expressions to match against a module name
+to determine if it is allowed to be unused
+
+Please create an [issue](https://github.com/charlierudolph/dependency-lint/issues)
+anytime you need to use this
+
+default:
+```yml
+allowUnused: []
```
Comment (kevgo) [inline docs/configuration.md]: I would say no need to provide the default values here. (A) to me "default" means what the system uses if this isn't provided here. So the default file should be empty. (B) The user sees the "default" values when she looks at the `dependency-lint.yml` file that is generated for her. 

What is more useful here in terms of documenting things would be some representative examples. With those, you can omit describing the type in the comments. This (A) makes the comments shorter and more readable, and (B) gives some practical examples how exactly it looks like if I want to provide values here.

```
### allowUnused

List of modules that should be ignored by dependency-lint.

Please create an [issue](https://github.com/charlierudolph/dependency-lint/issues)
anytime you need to use this

Examle:

allowUnused:
  - mocha
```

File: src/cli.coffee [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
     console.log 'Configuration file generated at "dependency-lint.yml"'
-  fs.copy src, dest, callback
+  async.waterfall [
+    (next) -> fs.readFile src, 'utf8', next
+    (contents, next) ->
+      comment = """
+        # See #{packageJson.homepage}/blob/v#{packageJson.version}/docs/configuration.md
+        # for a detailed explanation of the options
+        """
+      fs.writeFile dest, comment + '\n\n' + contents, next
```
Comment (kevgo) [inline src/cli.coffee]: nitpick: use string interpolation here

``` coffeescript
fs.writeFile dest, "#{comment}\n\n#{contents}", next
```

Comment (kevgo) [inline src/cli.coffee]: Actually, why don't you use the multi-line string that is currently assigned to `comment` as your template?

``` coffeescript
(contents, next) ->
  fileContent = """
    # See #{packageJson.homepage}/blob/v#{packageJson.version}/docs/configuration.md
    # for a detailed explanation of the options

    #{contents}
    """
  fs.writeFile dest, fileContent, next
```

File: docs/configuration.md (hunk 1 above)
Comment (charlierudolph) [inline docs/configuration.md]: The user will lose the default if they decide to remove the value from the config (or if its a new option and thus not listed in their config). I could just link to the default config from this file.

Comment (kevgo) [inline docs/configuration.md]: Yes, linking to the default values sounds like a good solution. Then this file can focus solely on providing useful documentation and examples.

**Comment by charlierudolph** [comment]
@kevgo updated

File: docs/configuration.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+Example:
+```yml
+allowUnused:
+  - mocha
+```
+
+---
+### devFilePatterns
+Files used only for development.
+Uses [minimatch](https://github.com/isaacs/minimatch)
```
Comment (kevgo) [inline docs/configuration.md:22]: I always wondered whether we need to mention the particular library that parses this here. I think everybody understands glob patterns pretty well, especially with the example.

File: docs/configuration.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+Example:
+```yml
+devFilePatterns:
+  - '**/*_test.js'
+```
+
+---
+### devScripts
+List of scripts in your `package.json` used only for development.
+Passed to `string.match`
```
Comment (kevgo) [inline docs/configuration.md:33]: Same here. I think if you provide a second example that is a regex this is appropriately covered here. I don't even know of the top of my hat what `string.match` does, I would have to look this up.

**Comment by kevgo** [comment]
Looks good to me! Feel free to :ship: at any time, or we can philosophize a bit more about fine-tuning the options. Happy to take this as far as you want or have time for.

File: docs/configuration.md (hunk 3 above)
Comment (charlierudolph) [inline docs/configuration.md:22]: I'd rather be more explicit then less. Glob implementations are not all the same.

File: docs/configuration.md (hunk 4 above)
Comment (charlierudolph) [inline docs/configuration.md:33]: I don't think having to look it up is a bad thing. This is just for reference to know whats possible

File: docs/configuration.md (hunk 3 above)
Comment (kevgo) [inline docs/configuration.md:22]: Maybe say "see [all syntax options]" with this being a link to the minimartch read me section that details the glob patterns? This way you still convey all the information (what parser you use, what the options are), with the text being less cluttered with technical implementation details.

**Pull request closed without merging**